lib/declare_schema.rb

```diff
@@ -39,6 +39,7 @@ def to_class(type)
 require 'declare_schema/field_declaration_dsl'
 require 'declare_schema/model'
 require 'declare_schema/model/field_spec'
-require 'declare_schema/model/index_spec'
+require 'declare_schema/model/index_definition'
+require 'declare_schema/model/foreign_key_definition'
```

Comment:
I renamed both of these from _spec to _definition so they wouldn't be confused with the tests that also end in _spec.
Note that field_spec should be changed like that someday. But also it should probably become column_spec or attribute_spec anyway.
Also, I split the foreign_key_definition out into its own file, as is best practice.